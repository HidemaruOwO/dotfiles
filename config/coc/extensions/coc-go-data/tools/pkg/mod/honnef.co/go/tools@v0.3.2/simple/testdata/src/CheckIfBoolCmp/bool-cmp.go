package pkg

func fn1() bool { return false }
func fn2() bool { return false }

func fn() {
	type T bool
	var x T
	const t T = false
	if x == t {
	}
	if fn1() == true { //@ diag(`simplified to fn1()`)
	}
	if fn1() != true { //@ diag(`simplified to !fn1()`)
	}
	if fn1() == false { //@ diag(`simplified to !fn1()`)
	}
	if fn1() != false { //@ diag(`simplified to fn1()`)
	}
	if fn1() && (fn1() || fn1()) || (fn1() && fn1()) == true { //@ diag(`simplified to (fn1() && fn1())`)
	}

	if (fn1() && fn2()) == false { //@ diag(`simplified to !(fn1() && fn2())`)
	}

	var y bool
	for y != true { //@ diag(`simplified to !y`)
	}
	if !y == true { //@ diag(`simplified to !y`)
	}
	if !y == false { //@ diag(`simplified to y`)
	}
	if !y != true { //@ diag(`simplified to y`)
	}
	if !y != false { //@ diag(`simplified to !y`)
	}
	if !!y == false { //@ diag(`simplified to !y`)
	}
	if !!!y == false { //@ diag(`simplified to y`)
	}
	if !!y == true { //@ diag(`simplified to y`)
	}
	if !!!y == true { //@ diag(`simplified to !y`)
	}
	if !!y != true { //@ diag(`simplified to !y`)
	}
	if !!!y != true { //@ diag(`simplified to y`)
	}
	if !y == !false { // not matched because we expect true/false on one side, not !false
	}

	var z interface{}
	if z == true {
	}
}
