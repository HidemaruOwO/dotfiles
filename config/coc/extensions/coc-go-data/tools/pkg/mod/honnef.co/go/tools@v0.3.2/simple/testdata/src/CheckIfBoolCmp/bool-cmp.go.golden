package pkg

func fn1() bool { return false }
func fn2() bool { return false }

func fn() {
	type T bool
	var x T
	const t T = false
	if x == t {
	}
	if fn1() { //@ diag(`simplified to fn1()`)
	}
	if !fn1() { //@ diag(`simplified to !fn1()`)
	}
	if !fn1() { //@ diag(`simplified to !fn1()`)
	}
	if fn1() { //@ diag(`simplified to fn1()`)
	}
	if fn1() && (fn1() || fn1()) || (fn1() && fn1()) { //@ diag(`simplified to (fn1() && fn1())`)
	}

	if !(fn1() && fn2()) { //@ diag(`simplified to !(fn1() && fn2())`)
	}

	var y bool
	for !y { //@ diag(`simplified to !y`)
	}
	if !y { //@ diag(`simplified to !y`)
	}
	if y { //@ diag(`simplified to y`)
	}
	if y { //@ diag(`simplified to y`)
	}
	if !y { //@ diag(`simplified to !y`)
	}
	if !y { //@ diag(`simplified to !y`)
	}
	if y { //@ diag(`simplified to y`)
	}
	if y { //@ diag(`simplified to y`)
	}
	if !y { //@ diag(`simplified to !y`)
	}
	if !y { //@ diag(`simplified to !y`)
	}
	if y { //@ diag(`simplified to y`)
	}
	if !y == !false { // not matched because we expect true/false on one side, not !false
	}

	var z interface{}
	if z == true {
	}
}
